unattended_installer/install_functions/opendistro/common.sh

```diff
@@ -274,107 +274,86 @@ function restoreWazuhrepo() {
+            uninstallmanager
+        else
+            logger "Wazuh manager components were not found on the system so it was not uninstalled."
+        fi
     fi
+    if [ -n "${AIO}" ] || [ "${uninstall_component_name}" == "all" ] || [ "${uninstall_component_name}" == "elasticsearch" ]; then
+        if [ -n "${elasticsearchinstalled}" ] || [ -n "${elastic_remaining_files}" ]; then
+            uninstallelasticsearch
+        else
 
-    if [[ ( -n "${wazuh_remaining_files}"  || -n "${wazuhinstalled}" ) && ( -n "${wazuh}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        eval "rm -rf /var/ossec/ ${debug}"
+            logger "Elasticsearch components were not found on the system so it was not uninstalled."
+        fi
     fi
-
-    if [[ -n "${elasticsearchinstalled}" && ( -n "${elasticsearch}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        logger -w "Removing Elasticsearch."
-        if [ "${sys_type}" == "yum" ]; then
-            eval "yum remove opendistroforelasticsearch -y ${debug}"
-            eval "yum remove elasticsearch* -y ${debug}"
-            eval "yum remove opendistro-* -y ${debug}"
-        elif [ "${sys_type}" == "zypper" ]; then
-            eval "zypper -n remove opendistroforelasticsearch elasticsearch* opendistro-* ${debug}"
-        elif [ "${sys_type}" == "apt-get" ]; then
-            eval "apt remove --purge ^elasticsearch* ^opendistro-* ^opendistroforelasticsearch -y ${debug}"
-        fi 
+    if [ -n "${AIO}" ] || [ ${uninstall_component_name} == "all" ] || [ ${uninstall_component_name} == "kibana" ] ; then
+        if [ -n "${kibanainstalled}" ] || [ -n "${kibana_remaining_files}" ]; then
+            uninstallkibana
+        else
+            logger "Kibana components were not found on the system so it was not uninstalled."
+        fi
     fi
 
-    if [[ ( -n "${elastic_remaining_files}" || -n "${elasticsearchinstalled}" ) && ( -n "${elasticsearch}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        eval "rm -rf /var/lib/elasticsearch/ ${debug}"
-        eval "rm -rf /usr/share/elasticsearch/ ${debug}"
-        eval "rm -rf /etc/elasticsearch/ ${debug}"
+    # Overwrite case
+    if [ -n "${overwrite}" ] && [ -n "${wazuh}" ]; then
+        uninstallmanager
+    fi
+    if [ -n "${overwrite}" ] && [ -n "${elasticsearch}" ]; then
+        uninstallelasticsearch
+    fi
+    if [ -n "${overwrite}" ] && [ -n "${kibana}" ]; then
+        uninstallkibana
     fi
 
-    if [[ -n "${filebeatinstalled}" && ( -n "${wazuh}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        logger -w "Removing Filebeat."
-        if [ "${sys_type}" == "yum" ]; then
-            eval "yum remove filebeat -y ${debug}"
-        elif [ "${sys_type}" == "zypper" ]; then
-            eval "zypper -n remove filebeat ${debug}"
-        elif [ "${sys_type}" == "apt-get" ]; then
-            eval "apt remove --purge filebeat -y ${debug}"
+    # rollBack case!
```

Comment:
Why the `!`? 
```suggestion
    # rollBack case
```

Reply:
Done https://github.com/wazuh/wazuh-packages/pull/1211/commits/c3fc83525f9efa17705f53a3ff6e1f8d681d1d49